save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProjectController.kt

```diff
@@ -168,69 +178,94 @@ class DownloadProjectController(private val configProperties: ConfigProperties)
-                    .doOnNext { testSuiteList ->
-                        makeRequest(BodyInserters.fromValue(getAllTests(resourcesRootPath, testSuiteList)), "/initializeTests?executionId=$executionId") { it.toBodilessEntity() }
-                            .doOnNext {
-                                // Post request to orchestrator to initiate its work
-                                log.debug("Knock-Knock Orchestrator")
-                                webClientOrchestrator
-                                    .post()
-                                    .uri("/initializeAgents")
-                                    .body(BodyInserters.fromValue(execution.also { it.id = execId }))
-                                    .retrieve()
-                                    .toEntity(HttpStatus::class.java)
-                                    .subscribe()
-                            }.subscribe()
-                    }.subscribe()
-            }.subscribe()
+            ExecutionStatus.PENDING, "ALL", resourcesRootRelativePath, 0, configProperties.executionLimit, executionType)
+        webClientBackend.makeRequest(BodyInserters.fromValue(execution), "/createExecution") { it.bodyToMono<Long>() }
+            .flatMap { executionId ->
+                val testResourcesRootAbsolutePath = getTestResourcesRootAbsolutePath(propertiesRelativePath, resourcesRootRelativePath)
+                discoverAndSaveTestSuites(project, testResourcesRootAbsolutePath, propertiesRelativePath)
+                    .flatMap { testSuites ->
+                        initializeTests(testSuites, testResourcesRootAbsolutePath, executionId)
+                            .then(initializeAgents(execution, executionId))
+                    }
+            }
+            .subscribe()
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun getTestResourcesRootAbsolutePath(propertiesRelativePath: String,
+                                                 resourcesRootRelativePath: String): String {
+        val saveProperties: SaveProperties = decodeFromPropertiesFile(
+            File(configProperties.repository, resourcesRootRelativePath)
+                .resolve(propertiesRelativePath)
+        )
+        return File(configProperties.repository, resourcesRootRelativePath)
+            .resolve(saveProperties.testConfigPath!!)
+            .absolutePath
+    }
+
+    private fun discoverAndSaveTestSuites(project: Project,
+                                          testResourcesRootPath: String,
+                                          propertiesRelativePath: String): Mono<List<TestSuite>> {
+        val testSuites: List<TestSuiteDto> = try {
+            testDiscoveringService.getAllTestSuites(project, testResourcesRootPath, propertiesRelativePath)
+        } catch (iae: IllegalArgumentException) {
+            log.error("Couldn't discover test suites, aborting: ", iae)
+            throw iae
```

Comment:
Why do you throw an exception?

Reply:
Removed in favor of common error handler